### DISCUSSION TITLE
Hierarchical Menu

### DISCUSSION THREAD

**Discussion by trioangle27** [algolia/instantsearch#5141] [General]
i am using hierarchical-menu option.to me category image and category title It should be a show.But  now current category title or category image only showing.but category image and title it should be show.

**Comment by Haroenv** [comment]
Hierarchical Menu only works on text. You will have to somehow create a mapping first:

```js
const mapping = {
    Audio: 'https://...',
   'Audio > Headphones': 'https://...',
};

// this can then be used in a custom rendering

const image = mapping[item.label]
```

This mapping can be created in many ways, one among which is a javascript object

**Comment by trioangle27** [comment]
thank you for response..
How to create mapping for algolia instatsearch.js

i am  using code:

instantsearch.widgets.hierarchicalMenu({
        container: '#header_category_image_algo',
        attributes: [
            'hierarchicalCategories.lvl0',
            'hierarchicalCategories.lvl1',
            'hierarchicalCategories.lvl2',
            'hierarchicalCategories.lvl3',
        ],
});

**Reply by Haroenv** [reply]
The mapping is something you do manually, as well as using a custom widget to be able to display the image. Unfortunately we have no example of this, as it's not a builtin feature
